UPGRADE-6.4.md

```diff
@@ -1,12 +1,46 @@
 UPGRADE FROM 6.3 to 6.4
 =======================
 
+DependencyInjection
+-------------------
+
+ * Deprecate `ContainerAwareInterface` and `ContainerAwareTrait`, use dependency injection instead
+
+   Before:
+   ```php
+   class MyService implements ContainerAwareInterface
+   {
+       use ContainerAwareTrait;
+
+       // ...
+   }
+   ```
+
+   After:
+   ```php
+   class MyService
+   {
+       // Inject the container through the constructor...
+       public function __construct(private ContainerInterface $container)
+       {
+       }
+
+       // ... or by using the #[Required] attribute
+       #[Required]
+       public function setContainer(ContainerInterface $container): void
+       {
+            $this->container = $container;
+       }
+   }
+   ```
```

Comment:
Might be worth mentioning that injecting the DIC itself is discouraged

Reply:
maybe we should drop those examples? I don't think it's common to use them

Reply:
I'd be fine dropping the examples. Injecting the container itself _seems_ to mainly occur in some legacy projects, I didn't see myself places where the container is directly injected in "recent" codebases.

About discouraging injecting the container, maybe it would be better to mention it somewhere in the documentation?

Reply:
Dropping the example sounds good enough. It has been said a lot already :)

Reply:
Dropped :+1: